### PR TITLE
Cleanup warnings in tests

### LINKED ISSUE #1805: lots of warnings creeping back into tests
after #1538, the tests summary was clean no warnings.  looks like a lot of warnings have crept back into the tests (particularly from the shapes deprecation warning in #1771).  just a reminder that the `[tool.pytest.ini_options]` section of `pyproject.toml` will let you batch ignore warnings if you know they are negligible.

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
closes #1805 by clearing up warnings again.  This time, I've added the filterwarning `"error:::napari"`... which will turn all warnings originating from within napari into errors unless explicitly allowed using `with pytest.warns(...)`.

### CODE REVIEW

**Comment by sofroniewn** [comment]
This is very interesting, our pre-test is now failing because of what looks like some numpy changes deprecating something! We've also got a magicgui segfault unrelated to this PR and another fail. We'll probably have to #2045 fail too if I rerun.

Overall I'm good with this PR, we don't want uncaught warnings floating around, but we'll probably want to get the CI back up to 100% passing before final review and merge

**Comment by tlambert03** [comment]
> but we'll probably want to get the CI back up to 100% passing before final review and merge

yeah, i agree.  sounds good

**Comment by sofroniewn** [comment]
I've just updated from master, but it looks like we've got a couple tests now failing specifically on this PR. Havn't looked at details yet

**Comment by tlambert03** [comment]
they seem to be mostly just our own warnings that don't normally get triggered.  should be easily fixed, but will take a look

**Comment by tlambert03** [comment]
you know, an alternative to turning all of our "unasserted" internal warnings into errors (i.e. being explicit about testing our warnings), is simply auto-ignoring them all (i.e. saying, it's always fine if we emit warnings ourself... its the ones that we didn't expect that we want to find).  thoughts?

**Comment by sofroniewn** [comment]
> you know, an alternative to turning all of our "unasserted" internal warnings into errors (i.e. being explicit about testing our warnings), is simply auto-ignoring them all (i.e. saying, it's always fine if we emit warnings ourself... its the ones that we didn't expect that we want to find). thoughts?

I'm open to that, but I also didn't mind turning ours into errors. We add a lot of warnings now during deprecations and so don't want to be accidentally using those methods/ properties still internally. This could help catch that, which I think was the original motivation. I'd defer to @jni if he had a strong opinion though, I don't really mind either way

**Comment by tlambert03** [comment]
ok, will leave them in and just "allow" the warnings that come from really early versions of stuff.  the "--pre" warnings are more interesting, and I need to look at them

**Comment by tlambert03** [comment]
ready again @sofroniewn.  main changes from last time:
- fixing all of the deprecation warnings from numpy 1.20 (e.g. `np.float -> float`)
- warnings are _only_ errors if they come from a `napari.` module (which excludes plugin `napari_*` modules)
- any `vendored` modules within napari are exempt
- whitelisted the "update your dask" warning triggered in the MIN_REQs check

**Review by sofroniewn — APPROVED** [review]
Looks good to me, thanks @tlambert03! We can continue to adjust the `filterwarnings` as needed, but I think this puts us in a good place.

I will leave open for a bit in case anyone else wants to comment, but will then merge!

File: pyproject.toml
Comment (sofroniewn) [inline pyproject.toml:77]: I'm trying to find a link explaining the `:::` syntax - i.e. why 3 colons but couldn't. I can find the warnings -> error here https://docs.pytest.org/en/stable/warnings.html, but not the bit about making it only from one package. Maybe I just missed it. If you have a link can you respond to this comment with it. No need to add it to the PR, but I'm just curious to understand a bit more about the syntax. Thanks!!

Comment (tlambert03) [reply pyproject.toml:77]: Those are warning filter strings from the standard library: https://docs.python.org/3/library/warnings.html#describing-warning-filters  I'll add a link

Comment (sofroniewn) [reply pyproject.toml:77]: ah love it! thanks for the link!!

Comment (tlambert03) [reply pyproject.toml:77]: comment added

**Pull request merged**
